rov/movement/controls/Movement_Algorithm.py

```diff
@@ -0,0 +1,98 @@
+from PID_Controller import PID
+from Base_Algorithm import Algorithm
+import time
+
+# Control Algorithm
+# README:
+#   Initalize with the parameter and sensor data
+#   Parameter being either 'x', 'y', 'z', 'roll', 'pitch' or 'yaw'
+#   Data from the sensors containing pressure or IMU
+#   The calculate function uses change in time, current position, desired position as inputs for the PID controller.
+#   This then returns a 6 degree output as the recommended user input to best achieve the desired position
+#   This algorithm has the option to activate and deactivate itself with the default being deactivated
+#   If deactived it always returns an empty output of the six degrees of freedom [0,0,0,0,0,0]
+#   This also allows tuning of the PID values by using @property to change and refer to the PID values of the controller.
+# How to used Control Algorithm class:
+# 1. control = ControlAlorithm('roll')
+#   -Initially deactivated
+# 2. control.activate()
+# 3. output = control.calculate()
+# 4. output will now contain an array of the suggested output
+# ex: [0, 0, 0, 0.5, 0, 0]
+# - If you wish to turn off the algorithm use: control.deactivate()
+# - This will cause calculate() to return an empty array: [0, 0, 0, 0, 0, 0]
+# - If you wish to turn back on use: control.activate()
+# - You can also use .toggle() to toggle between activated and deactivated
+# - For PID tuning you can directly get and set the values using .p, .i, .d
+# - ex: control.p = 5 or control.d = 2
+
+
+class MovementAlgorithm(Algorithm):
```

Comment:
Come up with a better name for ControlAlgorithm and MovementAlgorithm... We're talking about maintaining speed and maintaining position, name them with that relavent information.